### PR TITLE
Integrate remote config for updates

### PR DESCRIPTION
**Pull request by aidaiym**
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: app/lib/core/remote_config/mq_remote_config.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  int get recommendedBuildNumber => version.$2;
+
+  (int, int) get version {
+    final data = jsonDecode(remoteConfig.getString(_appVersion)) as Map<String, dynamic>;
+    final platformName = Platform.isIOS ? 'ios' : 'android';
+    final versionData = data[platformName] as Map<String, dynamic>;
+    return (versionData['requiredBuildNumber'], versionData['recommendedBuildNumber']);
+  }
+
+  static const _disableHatim = 'isHatimDisabled';
```
Comment (Eldar2021) [inline app/lib/core/remote_config/mq_remote_config.dart:40]: I suggest rename `_hatimIsEnable`;

File: app/lib/modules/home/presentation/view/home_view.dart [hunk 2]
```diff
@@ -103,8 +103,14 @@ class HomeBody extends StatelessWidget {
...
-                  MqAnalytic.track(AnalyticKey.goHatim);
-                  context.goNamed(AppRouter.hatim);
+                  if (context.read<MqRemoteConfig>().isHatimDisabled) {
+                    ScaffoldMessenger.of(context).showSnackBar(
+                      SnackBar(content: Text(context.l10n.hatimNotAvailable)),
+                    );
+                  } else {
+                    MqAnalytic.track(AnalyticKey.goHatim);
+                    context.goNamed(AppRouter.hatim);
+                  }
```
Comment (Eldar2021) [inline app/lib/modules/home/presentation/view/home_view.dart:113]: I doubt if it's working properly...

File: app/test/helpers/pump_app.dart [hunk 3]
```diff
@@ -47,6 +49,10 @@ extension PumpApp on WidgetTester {
             BlocProvider(
               create: (context) => HomeCubit(GetHomeDataUseCase(homeRepo)),
             ),
+            BlocProvider(
+              create: (context) => RemoteConfigCubit(),
+            ),
+            RepositoryProvider<MqRemoteConfig>(create: (context) => remoteConfig),
```
Comment (Eldar2021) [inline app/test/helpers/pump_app.dart:55]: RepositoryProvider in MultiBlocProvider? I think this not good way. Please use MultiRepositoryProvider

```dart
    return pumpWidget(
      MultiRepositoryProvider(
        providers: [
          RepositoryProvider(create: (context) => const AppConfig(isIntegrationTest: true)),
          RepositoryProvider<MqRemoteConfig>(create: (context) => remoteConfig),
        ],
        child: MultiBlocProvider(
          providers: [
            BlocProvider(
              create: (context) => AppCubit(
                getInitialThemeUseCase: getInitialThemeUseCase,
                setModeUseCase: setModeUseCase,
                setColorUseCase: setColorUseCase,
                getAppVersionUseCase: getAppVersionUseCase,
              ),
            ),
            BlocProvider(
              create: (context) => AuthCubit(
                getInitialUserUseCase,
                googleSignInUseCase,
                appleSignInUseCase,
                setUserDataUseCase,
                patchGenderUseCase,
                patchLocaleCodeUseCase,
                logoutUseCase,
              ),
            ),
            BlocProvider(create: (context) => HomeCubit(GetHomeDataUseCase(homeRepo))),
            BlocProvider(create: (context) => RemoteConfigCubit()),
          ],
          child: const QuranApp(),
        ),
      ),
    );
    
    ```

File: app/test/widget_test.dart [hunk 4]
```diff
@@ -28,13 +29,47 @@ final class MockHomeRepositoryImpl implements HomeRepository {
   }
 }
 
+final class MockFirebaseRemoteConfig extends Mock implements FirebaseRemoteConfig {
```
Comment (Eldar2021) [inline app/test/widget_test.dart:32]: Very nice solution 👍

**Review by Eldar2021 — COMMENTED** [review]
I am happy to review with a good PR again.
But I still think we have some problems.
Please use `MqRemoteConfig from `RemoteConfigCubit`, and free `MqRemoteConfig` from `buildNumber` dependency. Give `PackageInfo` to `RemoteConfigCubit` and get `currentBuildNumber` from there. Also use BlocBuilder for the hatin button and get the value from `RemoteConfigCubit`. 
Finally it is best to run ` _listenRemoteConfig()` from `RemoteConfigCubit`.

Translated with DeepL.com (free version).

File: app/lib/modules/home/presentation/view/home_view.dart (hunk 2 above)
Comment (aidaiym) [reply app/lib/modules/home/presentation/view/home_view.dart:113]: why?

File: app/lib/app/presentation/view/app_view.dart [hunk 5]
```diff
@@ -78,26 +79,48 @@ class MyApp extends StatelessWidget {
...
+}
+
+class _QuranAppState extends State<QuranApp> {
+  @override
+  void initState() {
+    super.initState();
+    SchedulerBinding.instance.addPostFrameCallback((_) {
+      context.read<RemoteConfigCubit>().init();
+    });
+  }
```
Comment (Eldar2021) [inline app/lib/app/presentation/view/app_view.dart:108]: Cool!

**Review by Eldar2021 — APPROVED** [review]
It works exactly as I expected. Thank you very much!
I brought some changes please review:) https://github.com/Eldar2021/my_quran/pull/224/commits/91dfb482f29823bcbd50090e05753a6b7ffb6252

**Pull request merged**